File: app/src/main/java/com/kuyuzhiqi/baseproject/network/HttpService.java
package com.kuyuzhiqi.baseproject.network;

import com.kuyuzhiqi.baseproject.network.response.LoginResponse;
import com.kuyuzhiqi.bizbase.entity.RequestPortBO;
import com.kuyuzhiqi.network.HttpFacade;
import io.reactivex.Single;
import io.reactivex.schedulers.Schedulers;
import  com.kuyuzhiqi.bizbase.network.ConvertBizResponseForSingle;

import java.util.TreeMap;

public class HttpService {
    private String mHost;
    private HttpService instance;
    private Api api;

    public void register(String host) {
        mHost = host;
        instance = new HttpService();
        api = new HttpFacade(host).createApi(Api.class);
    }

    private static TreeMap<String, String> buildDefaultParamMap() {
        TreeMap<String, String> treeMap = new TreeMap<>();
        treeMap.put("device_id", "deviceId");
        return treeMap;
    }

    public Single<LoginResponse> applyLoginIn(String userName, String password) {
        RequestPortBO requestPortBO = new RequestPortBO(Api.LOGIN_IN_PORT, mHost, Api.LOGIN_IN_URL);
        TreeMap<String, String> treeMap = buildDefaultParamMap();
        treeMap.put("userName", userName);
        treeMap.put("password", password);
        requestPortBO.setParamMap(treeMap);
        return api.doLogin(treeMap)
                .subscribeOn(Schedulers.io())
                .compose(new ConvertBizResponseForSingle(requestPortBO));
    }

}

File: app/src/main/java/com/kuyuzhiqi/baseproject/network/HttpService.kt
package com.kuyuzhiqi.baseproject.network

import com.kuyuzhiqi.baseproject.network.response.LoginResponse
import com.kuyuzhiqi.bizbase.entity.RequestPortBO
import com.kuyuzhiqi.network.HttpFacade
import io.reactivex.Single
import io.reactivex.schedulers.Schedulers
import com.kuyuzhiqi.bizbase.network.ConvertBizResponseForSingle

import java.util.TreeMap

class HttpService {
    private var mHost: String? = null
    private var instance: HttpService? = null
    private var api: Api? = null

    fun register(host: String) {
        mHost = host
        instance = HttpService()
        api = HttpFacade(host).createApi(Api::class.java)
    }

    private fun buildDefaultParamMap(): TreeMap<String, String> {
        val treeMap = TreeMap<String, String>()
        treeMap["device_id"] = "deviceId"
        return treeMap
    }

    fun applyLoginIn(userName: String, password: String): Single<LoginResponse> {
        val requestPortBO = RequestPortBO(Api.LOGIN_IN_PORT, mHost, Api.LOGIN_IN_URL)
        val treeMap = buildDefaultParamMap()
        treeMap["userName"] = userName
        treeMap["password"] = password
        requestPortBO.paramMap = treeMap
        return api!!.doLogin(treeMap)
            .subscribeOn(Schedulers.io())
            .compose(ConvertBizResponseForSingle(requestPortBO))
    }

}
